internal/internal.go

```diff
@@ -267,3 +267,9 @@ const (
 // It currently has an experimental suffix which would be removed once
 // end-to-end testing of the policy is completed.
 const RLSLoadBalancingPolicyName = "rls_experimental"
+
+// EnforceSubConnEmbedding is used to enforce proper SubConn implementation
+// embedding.
+type EnforceSubConnEmbedding interface {
+	enforceSubConnEmbedding()
+}
```

Comment:
I see, extracting the private method in a new interface ensures compilation fails if `acBalancerWrapper` and other SubConn implementations are missing a SubConn method. 

Should we also embed either the internal interface or the NoOpSubConn in [TestSubConn](https://github.com/grpc/grpc-go/blob/a365199cb21f2bcf8fc46a4f3edb87f9adfab637/internal/testutils/balancer.go#L35) and [testSubConn](https://github.com/grpc/grpc-go/blob/a365199cb21f2bcf8fc46a4f3edb87f9adfab637/balancer/base/balancer_test.go#L44).

Reply:
This is where I paused, actually.  Maybe we don't want `NopSubConn`..  The question is whether it's better to provide something that silently does the wrong thing because something is not implemented correctly, but is needed, or recommend embedding the interface and not setting it, making it panic instead.

I kind of think it should be the latter.  Iterating.